tests/tools.py

```diff
@@ -0,0 +1,209 @@
+    ),
+]
+
+NUM_EVENTS = len(dummy_events)
+NUM_PATIENTS = 10
+
+all_events: List[Tuple[int, piton.Event]] = []
+
+for patient_id in range(10, 10 + NUM_PATIENTS):
+    all_events.extend((patient_id, event) for event in dummy_events)
+
+
+def create_events(tmp_path: pathlib.Path) -> piton.datasets.EventCollection:
+    events = piton.datasets.EventCollection(os.path.join(tmp_path, "events"))
+    chunks = 7
+    events_per_chunk = (len(all_events) + chunks - 1) // chunks
+
+    for i in range(7):
+        with contextlib.closing(events.create_writer()) as writer:
+            for patient_id, event in all_events[
+                i * events_per_chunk : (i + 1) * events_per_chunk
+            ]:
+                writer.add_event(patient_id, event)
+
+    return events
+
+
+def create_patients(tmp_path: pathlib.Path) -> piton.datasets.PatientCollection:
+    return create_events(tmp_path).to_patient_collection(
+        os.path.join(tmp_path, "patients")
+    )
+
+
+def create_ontology(path_to_ontology_dir: str, concepts: List[str]):
+    path_to_concept_file: str = os.path.join(
+        path_to_ontology_dir, "concept", "concept.csv.zst"
+    )
+    os.makedirs(os.path.dirname(path_to_concept_file), exist_ok=True)
+    os.makedirs(
+        os.path.join(path_to_ontology_dir + "/concept_relationship/"),
+        exist_ok=True,
+    )
+
+    concept_map: Dict[str, int] = {}
+
+    with io.TextIOWrapper(
+        zstandard.ZstdCompressor(1).stream_writer(
+            open(path_to_concept_file, "wb")
+        )
+    ) as o:
+        writer = csv.DictWriter(
+            o,
+            fieldnames=[
+                "concept_id",
+                "concept_name",
+                "domain_id",
+                "vocabulary_id",
+                "concept_class_id",
+                "standard_concept",
+                "concept_code",
+                "valid_start_DATE",
+                "valid_end_DATE",
+                "invalid_reason",
+                "load_table_id",
+                "load_row_id",
+            ],
+        )
+
+        writer.writeheader()
+
+        next_code: int = 0
+        for i, c in enumerate(concepts):
+            code: int = i + next_code
+            concept_map[c] = code
+            writer.writerow(
+                {
+                    "concept_id": str(code),
+                    "concept_name": c,
+                    "domain_id": "Observation",
+                    "vocabulary_id": "dummy",
+                    "concept_class_id": "Observation",
+                    "standard_concept": "",
+                    "concept_code": c,
+                    "valid_start_DATE": "1970-01-01",
+                    "valid_end_DATE": "2099-12-31",
+                    "invalid_reason": "",
+                    "load_table_id": "custom_mapping",
+                    "load_row_id": "",
+                }
+            )
+    return concept_map
+
+
+class DummyOntology:
```

Comment:
Why are we creating a dummy ontology class here? If you are already going through the trouble of creating a real PatientDatabase, why not use the associated real ontology for that database?

Reply:
removed ontology class. All we need is a list of concept really.